cfgov/ask_cfpb/documents.py

```diff
@@ -35,7 +37,7 @@ def prepare_url(self, instance):
         return instance.url
 
     class Index:
-        name = 'ask-cfpb'
+        name = name = f'{settings.ES_INDEX_PREFIX}ask-cfpb'
```

Comment:
It'd be nice not to have to duplicate this each time we define a new index. What would you think about doing something like (completely untested):

```py
# somewhere central
class EnvironmentSpecificIndex:
    @property
    def name(self):
        return settings.ES_INDEX_PREFIX + self.name_no_prefix

# each time we define an index
@registry.register_document
class MyDocument(Document):
    ...

    class Index(EnvironmentSpecificIndex):
        name_no_prefix = 'ask-cfpb'
        ...
```

This way we'd still have to remember to use EnvironmentSpecificIndex but maybe that's more easily documented and less error-prone than copying the prefix each time?

Reply:
Tried the class approach, but looks like the way the dsl library access the name attribute wasn't working with the property. Opted instead to move it into a central function `environment_specific_index`. The name is a working title, but I think this addresses reusability concerns. We also wanted both local and production indexes to have no prefix, so I added the little snippet of logic around handling that.